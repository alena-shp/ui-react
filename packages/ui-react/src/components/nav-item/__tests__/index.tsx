import React from 'react';
import { create } from 'react-test-renderer';

import { NavItem } from '..';

test('Default', () => {
    const tree = create(
        <NavItem
            href="#"
            children="Элемент"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Active', () => {
    const tree = create(
        <NavItem
            presetStyle="active"
            href="#active"
            children="Элемент"
            title="Тайтл"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Disable', () => {
    const tree = create(
        <NavItem
            presetStyle="active"
            href="#active"
            children="Элемент"
            title="Тайтл"
            isDisabled={true}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Download', () => {
    const tree = create(
        <NavItem
            presetStyle="active"
            href="#active"
            children="Элемент"
            title="Тайтл"
            download={true}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});
