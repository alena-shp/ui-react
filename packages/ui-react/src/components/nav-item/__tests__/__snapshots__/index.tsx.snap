// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active 1`] = `
<a
  className="nav-item nav-item_preset-active"
  href="#active"
  title="Тайтл"
>
  Элемент
</a>
`;

exports[`Default 1`] = `
<a
  className="nav-item nav-item_preset-default"
  href="#"
  title="Элемент"
>
  Элемент
</a>
`;

exports[`Disable 1`] = `
<a
  className="nav-item nav-item_preset-active nav-item_disabled"
  href="#active"
  title="Тайтл"
>
  Элемент
</a>
`;

exports[`Download 1`] = `
<a
  className="nav-item nav-item_preset-active"
  download={true}
  href="#active"
  title="Тайтл"
>
  Элемент
</a>
`;
